src/EnergyPlus/BaseboardRadiator.cc

```diff
@@ -643,13 +643,13 @@ namespace BaseboardRadiator {
                     }
                 } else {
                     CheckZoneSizing(state, cCMO_BBRadiator_Water, this->EquipID);
-                    std::string CompType = cCMO_BBRadiator_Water;
-                    std::string CompName = this->EquipID;
+                    std::string_view const CompType = cCMO_BBRadiator_Water;
+                    std::string_view const CompName = this->EquipID;
                     state.dataSize->DataFracOfAutosizedHeatingCapacity = 1.0;
                     state.dataSize->DataZoneNumber = this->ZonePtr;
                     int SizingMethod = DataHVACGlobals::HeatingCapacitySizing;
                     int FieldNum = 1;
-                    std::string SizingString = this->FieldNames(FieldNum) + " [W]";
+                    std::string_view SizingString = this->FieldNames(FieldNum) + " [W]";
```

Comment:
@amirroth I originally tried this syntax and the compiler does not like it. 

![image](https://github.com/NREL/EnergyPlus/assets/5778156/c1816c44-598a-4364-85e9-384c595af20c)

So instead of reverting back to std::string I thought I would try something else. Why doesn't the compiler like the 3rd and 4th attempt here?

![image](https://github.com/NREL/EnergyPlus/assets/5778156/7b3a8934-78c5-4e46-a4d5-59d2dfa4b19b)

Reply:
OK, so if the compiler wants a memory location then why not just provide one:

![image](https://github.com/NREL/EnergyPlus/assets/5778156/5d580a36-b886-46e2-94e2-2899d55b8ba3)

Reply:
Back to a string I guess:

![image](https://github.com/NREL/EnergyPlus/assets/5778156/5e03a7c2-b7f3-4b5e-80e3-8196dd29cc67)

Reply:
So, the terminology for what's going on here is a little weird but the concept is simple enough. Programming languages distinguish between things called l-values (values that can appear on either the left or the right side of an assignment operator, specifically they can be assigned to) and r-values (values that can only appear on the right side of an assignment, i.e., they cannot be assigned into). Things like expressions and function calls are r-values. You cannot do `(x + 1) = y;` or  `foo(x) = 5;`. The reason is that r-values are temporary, they only live within the expression, once they are used they are essentially gone, they are not permanently stored for use in subsequent statements. 

Ok, what does this have to do with the current situation? You cannot assign a `string_view` from an r-value that returns a `string`. When you have a function call or an expression that returns a new string, e.g., `format()` or `str + "[W]"` that new string does not live permanently, it is either copied into a permanent location (i.e., a string l-value) or it disappears. A `string_view`, however, has no storage for you to copy the string r-value into. And therefore you should only assign a `string_view` from an l-value, i.e., a string that already has its own permanent storage. 

Incidentally, this is the same exact "rule" for using `std::string` vs. `std::string const &`. You cannot assign a `std::string const &` from an expression like `str + "[W]"`. That expression is not a permanent string, so what is the reference referencing? You should also not assign a `std::string const &` from a function that returns a new string, e.g., `format`. However, if a function returns a reference to a permanent string, you can assign that to a `std::string const &`. 

Sorry for the "lecture", but that's what's going on here.

Reply:
That's a good explaination. I would have hated to miss that lecture.